src/Microsoft.DotNet.XHarness.WebAssembly/xunit-runner.cs

```diff
@@ -0,0 +1,349 @@
+						Console.WriteLine ("FAIL (discoverer): " + args [0] + " " + args [1]);
+						failed = true;
+					}
+
+					IDataDiscoverer discoverer;
+					discoverer = ExtensibilityPointFactory.GetDataDiscoverer (this, discovererType);
+
+					try {
+						var data = discoverer.GetData (dataAttribute, tc.TestMethod.Method);
+						var data_arr = data.ToArray ();
+						Console.WriteLine (tc.DisplayName + " [" + data_arr.Length + "]");
+						foreach (var dataRow in data_arr) {
+							nrun ++;
+							object obj = null;
+							if (!method.IsStatic) {
+								var constructor = method.ReflectedType.GetConstructor (Type.EmptyTypes);
+								if (constructor != null) 
+									obj = constructor.Invoke (null);
+								else
+									obj = System.Runtime.Serialization.FormatterServices.GetUninitializedObject (method.ReflectedType);
+							}
+
+							var pars = method.GetParameters ();
+							for (int i = 0; i < dataRow.Length; ++i)
+								dataRow [i] = ConvertArg (dataRow [i], pars [i].ParameterType);
+							method.Invoke (obj, BindingFlags.Default, null, dataRow, null);
+						}
+					} catch (Exception ex) {
+						Console.WriteLine ("FAIL: " + tc.DisplayName + " " + ex);
+						failed = true;
+					}
+				}
+				if (failed) {
+					nfail ++;
+					continue;
+				}
+			} else {
+				if (!String.IsNullOrEmpty (tc.SkipReason)) {
+					nskipped ++;
+					continue;
+				}
+				nrun ++;
+				var name = tc.DisplayName;
+				if (name.Contains ("("))
+					name = name.Substring (0, name.IndexOf ("("));
+				if (name != last_name)
+					Console.WriteLine (name);
+				last_name = name;
+				//Console.WriteLine (tc.DisplayName);
+				try {
+					object obj = null;
+					if (!method.IsStatic) {
+						var constructor = method.ReflectedType.GetConstructor (Type.EmptyTypes);
+						if (constructor != null) 
+							obj = constructor.Invoke (null);
+						else
+							obj = System.Runtime.Serialization.FormatterServices.GetUninitializedObject (method.ReflectedType);
+					}
+					var args = tc.TestMethodArguments;
+					if (args != null) {
+						var pars = method.GetParameters ();
+						for (int i = 0; i < args.Length; ++i) {
+							args [i] = ConvertArg (args [i], pars [i].ParameterType);
+						}
+					}
+					method.Invoke (obj, args);
+				} catch (Exception ex) {
+					Console.WriteLine ("FAIL: " + tc.DisplayName);
+					Console.WriteLine (ex);
+					nfail ++;
+				}
+			}
+		}
+
+		//foreach (var tc in testCases)
+		//	Console.WriteLine (tc.DisplayName);
+		//executor.RunTests (testCases, this, executionOptions);
+		Console.WriteLine ("TESTS = "  + testCases.Count + ", RUN = " + nrun + ", SKIP = " + nfiltered + ", FAIL = " + nfail);
+		return nfail == 0 ? 0 : 1;
+	}
+}
+
+class CmdLineParser : CommandLine
+{
+	public CmdLineParser (String[] args) : base (args, s => true) {
+	}
+}
+
+public class XunitDriver
+{
+	static WasmRunner testRunner;
+
+	static int Main (String[] args) {
+		// Process rsp files
+		// FIXME: This doesn't work with wasm
```

Comment:
Does mono was support mono options? You will get rap for free (I know @marek-safar might yell at me for typing this).

Reply:
Sorry, I don't get you here.

Reply:
you mean `MONO_ENV_OPTIONS`?  If so, yeah.